### DISCUSSION TITLE
Individual Options / Exclude a device type or role

### DISCUSSION THREAD

**Discussion by milevskiy2** [mattieserver/netbox-topology-views#483] [Ideas]
Good afternoon!

I have more than 700 devices in my network, including about 500 personal computers. 

Because of this, the scheme does not look very good. 

It would be nice to be able to exclude certain types of devices and/or roles through a filter.

**Comment by dreng** [comment]
You can go to the Filters pane and use filters for devices, device types, roles and many more in order to reduce the output. If you like to go the opposite way and exclude devices instead of including them, please see this FR: https://github.com/netbox-community/netbox-topology-views/issues/169

**Reply by milevskiy2** [reply]
Device type - there is no such filter
Device role - does not work
![1](https://github.com/netbox-community/netbox-topology-views/assets/167505163/dde0f301-9f82-48af-b68e-21e60909aff4)

**Comment by dreng** [comment]
The device type is called "Model" here. This corresponds to the Devices View in NetBox.

Role definitely works. I use it nearly every day. You may have to check "Show Unconnected" if the selected roles lead to a situation where no cable are connected between the filtered devices.

**Reply by milevskiy2** [reply]
If I select the "switch" role, it displays all the devices connected to it, including the Users' PCs. 

I need to exclude User PCs from the topology. How can I do this? 


https://github.com/netbox-community/netbox-topology-views/assets/167505163/ad448caf-0570-491e-8712-a5755b2d9a0b

**Reply by milevskiy2** [reply]
and even the "model" filter does not work perfectly. 

For some reason, some devices that are not in the list remain on the topology. 



https://github.com/netbox-community/netbox-topology-views/assets/167505163/813579f2-1274-407d-a0a5-4ef66eb45163

**Reply by dreng** [reply]
> If I select the "switch" role, it displays all the devices connected to it, including the Users' PCs.

It works as intended. You've selected "Show Neighbors".

> I need to exclude User PCs from the topology. How can I do this?

Again, please read #169.

> For some reason, some devices that are not in the list remain on the topology.

See above.

**Reply by milevskiy2** [reply]
Thank you. I figured it out, everything works as it should.
